core/src/main/scala/io/qbeast/core/model/CubeDomainsBuilder.scala

```diff
@@ -235,22 +204,18 @@ class CubeDomainsBuilder protected (
  * Factory for WeightAndCount. The creation of which depends on whether the associated CubeId is
  * present in the existing index, if it's a inner or leaf cube.
  *
- * @param existingNormalizedWeights
+ * @param existingCubeWeights
  *   Cube Weight of the existing index
- * @param groupCubeSize
- *   partition-level cube size for indexing
  */
-private class WeightAndCountFactory(
-    existingNormalizedWeights: Map[CubeId, NormalizedWeight],
-    groupCubeSize: Int) {
+private class WeightAndCountFactory(existingCubeWeights: Map[CubeId, Weight]) {
```

Comment:
Why Weight instead of NormalizedWeight? Could this affect calculations?

Reply:
Instead of computing `NormalizedWeight` and then using it to compute the `elementCount` for each existing leaf cube, we assume leaf cubes are empty.

This would place more elements in these leaf cubes during indexing. Still, as we are only interested in cube domains, this should be fine with the calculations.

Reply:
Let's test that this change does not compromise index layout with bigger workloads before introducing it 🙏